app/src/main/java/com/github/wnder/User.java

```diff
@@ -1,10 +1,37 @@
 package com.github.wnder;
 
+import android.Manifest;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.location.Location;
+import android.location.LocationListener;
+import android.location.LocationManager;
 import android.net.Uri;
+import android.os.Looper;
+import android.util.Log;
 
-public interface User {
+import androidx.annotation.NonNull;
+import androidx.core.app.ActivityCompat;
 
-    public String getName();
+import com.google.android.gms.maps.model.LatLng;
 
-    public Uri getProfilePicture();
+import java.util.IllegalFormatException;
+
+public abstract class User {
+
+    LatLng position = null;
+
+    public abstract String getName();
+
+    public abstract Uri getProfilePicture();
+
+    public Location getPositionFromGPS(LocationManager manager, Context context){
+        
+        if (ActivityCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(context, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+            //TODO: Maybe ask for permission? Or just consider it to be an impossible case
+            //throw new IllegalStateException();
+        }
```

Comment:
Maybe just return to the login activity ?

Reply:
It has to be discussed with the team, but for now it seems to be unreachable in the actual state of the app.